src/components/Card.js

```diff
@@ -1,26 +1,46 @@
 export default class Card {
-  constructor(data, cardSelector, handleImageClick) {
+  constructor(
+    data,
+    cardSelector,
+    handleImageClick,
+    handleDeleteButtonClick,
+    handleLikeButtonClick
+  ) {
     this._name = data.name;
     this._link = data.link;
+    this._id = data._id;
+    this._isLiked = data.isLiked;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleDeleteButtonClick = handleDeleteButtonClick;
+    this._handleLikeButtonClick = handleLikeButtonClick;
   }
   _setCardImageListeners() {
     this._cardImage.addEventListener("click", () => {
       this._handleImageClick(this._cardImage);
     });
   }
+  _checkLiked() {
+    if (this._isLiked) {
+      this._likeButton.classList.add("card__like-button_active");
+    }
+  }
+  _toggleLike() {
+    this._isLiked = !this._isLiked;
+    this._likeButton.classList.toggle("card__like-button_active");
+  }
   _setLikeButtonListeners() {
     this._likeButton.addEventListener("click", () => {
-      this._likeButton.classList.toggle("card__like-button_active");
+      this._handleLikeButtonClick(this._id, this._isLiked);
+      this._toggleLike();
```

Comment:
You need to **toggle likes** only in block `then` after a successful response from the server because if there is an error, the user will not know about that and will see the toggled heart color.